pkg/apis/core/v1alpha1/storagecluster.go

```diff
@@ -233,13 +233,35 @@ type StorageSpec struct {
 	RaidLevel *string `json:"raidLevel,omitempty"`
 }
 
+// CloudStorageCapacitySpec details the minimum and maximum amount of storage
+// that will be provisioned in the cluster for a particular set of minimum IOPS.
+type CloudStorageCapacitySpec struct {
+	// MinIOPS minimum IOPS expected from the cloud drive
+	MinIOPS uint32 `json:"minIOPS"`
+	// MinCapacityInGiB minimum capacity for this cloud device spec
+	MinCapacityInGiB uint64 `json:"minCapacity"`
+	// MaxCapacityInGiB capacity for this cloud device spec should not go above this threshold
+	MaxCapacityInGiB uint64 `json:"maxCapacity,omitempty"`
+	// Options additional options required to provision the drive in cloud
+	Options map[string]string `json:"options,omitempty"`
+}
+
 // CloudStorageSpec details of storage in cloud environment
 type CloudStorageSpec struct {
 	// DeviceSpecs list of storage device specs. A cloud storage device will
 	// be created for every spec in the DeviceSpecs list. Currently,
 	// CloudStorageSpec is only at the cluster level, so the below specs
 	// be applied to all storage nodes in the cluster.
 	DeviceSpecs *[]string `json:"deviceSpecs,omitempty"`
+
+	// CapacitySpecs list of cluster wide storage types and their capacities.
+	// A single capacity spec identifies a storage pool with a set of minimum
+	// requested IOPS and size. Based on the cloud provider, the total storage
+	// capacity will get divided amongst the nodes. The nodes bearing storage
+	// themselves will get uniformly distributed across all the zones.
+	// CapacitySpecs is slated to replace DeviceSpecs in v1alpha2 version of StorageCluster.
+	// CapacitySpecs []*CloudStorageCapacitySpec `json:"capacitySpecs,omitempty"`
```

Comment:
Just following the trend in this file. 
Should it be this instead ?
`CapacitySpecs *[]CloudStorageCapacitySpec`

Reply:
Most of the fields were made *type so they can be overridden in the node specs. It was to differentiate between empty list of devices and not specified. In case of capacity specs, it won't ever be a node specific spec. So it should be fine to leave it without pointer, right?

Reply:
Yes, makes sense.

Reply:
done